### PR TITLE
Clean up and modernize our C# code

### PR DESCRIPTION
**Pull request by pleroy**
Applied by hand some of the suggestions made by ReSharper.

### CODE REVIEW

File: ksp_plugin_adapter/flight_planner.cs [hunk 1]
```diff
@@ -145,14 +145,12 @@ class FlightPlanner : SupervisedWindowRenderer {
   private void RenderFlightPlan(string vessel_guid) {
     using (new UnityEngine.GUILayout.VerticalScope()) {
       if (final_time_.Render(enabled : true)) {
-        var status = plugin.FlightPlanSetDesiredFinalTime(vessel_guid,
-                                                          final_time_.value);
+        Status status = plugin.FlightPlanSetDesiredFinalTime(vessel_guid,
```
Comment (eggrobin) [inline ksp_plugin_adapter/flight_planner.cs]: why?

File: ksp_plugin_adapter/flight_planner.cs [hunk 2]
```diff
@@ -252,10 +250,11 @@ class FlightPlanner : SupervisedWindowRenderer {
...
           BurnEditor burn = burn_editors_[i];
-          if (burn.Render(header     : "Manœuvre #" + (i + 1),
-                          anomalous  : i >= (burn_editors_.Count -
-                                             number_of_anomalous_manœuvres),
-                          final_time : final_times[i])) {
+          if (burn.Render(header          : "Manœuvre #" + (i + 1),
+                          anomalous       :
+                              i >= (burn_editors_.Count -
+                                    number_of_anomalous_manœuvres),
+                          burn_final_time : final_times[i])) {
```
Comment (eggrobin) [inline ksp_plugin_adapter/flight_planner.cs]: This does not look like an improvement.

File: ksp_plugin_adapter/gl_lines.cs [hunk 3]
```diff
@@ -17,21 +17,19 @@ public enum Style {
...
+      Log.Fatal("Exception while drawing lines: " + e);
     }
   }
 
   public static void AddSegment(Vector3d world_begin,
                                 Vector3d world_end) {
-    var begin = WorldToMapScreen(world_begin);
-    var end = WorldToMapScreen(world_end);
+    UnityEngine.Vector3 begin = WorldToMapScreen(world_begin);
+    UnityEngine.Vector3 end = WorldToMapScreen(world_end);
```
Comment (eggrobin) [inline ksp_plugin_adapter/gl_lines.cs]: Why?

File: ksp_plugin_adapter/ksp_plugin_adapter.cs [hunk 4]
```diff
@@ -201,9 +200,9 @@ private enum UnityLayers {
   [KSPField(isPersistant = true)]
   internal MainWindow main_window_;
 
-  public event Action clear_locks;
-  public event Action dispose_windows;
-  public event Action render_windows;
+  public event Action ClearLocks;
+  public event Action DisposeWindows;
+  public event Action RenderWindows;
```
Comment (eggrobin) [inline ksp_plugin_adapter/ksp_plugin_adapter.cs]: Do we really want to change style for events? We had consistendly been using snake_case.

File: ksp_plugin_adapter/ksp_plugin_adapter.cs [hunk 5]
```diff
@@ -2205,7 +2190,7 @@ private enum UnityLayers {
     }
     must_set_plotting_frame_ = true;
   } catch (Exception e) {
-    Log.Fatal("Exception while resetting plugin: " + e.ToString());
+    Log.Fatal("Exception while resetting plugin: " + e);
```
Comment (eggrobin) [inline ksp_plugin_adapter/ksp_plugin_adapter.cs]: `$"Exception while resetting plugin: {e}"`

File: ksp_plugin_adapter/main_window.cs [hunk 6]
```diff
@@ -182,7 +182,7 @@ internal class MainWindow : SupervisedWindowRenderer {
               style: Style.Multiline(UnityEngine.GUI.skin.textArea));
         }
       }
-      Interface.GetVersion(build_date : out string unused_build_date,
+      Interface.GetVersion(build_date : out string _,
```
Comment (eggrobin) [inline ksp_plugin_adapter/main_window.cs]: Interesting convention.

File: ksp_plugin_adapter/window_renderer.cs [hunk 7]
```diff
@@ -90,11 +86,10 @@ internal class ScalingRenderer {
 }
 
 internal abstract class BaseWindowRenderer : ScalingRenderer, IConfigNode {
-  protected BaseWindowRenderer(UnityEngine.GUILayoutOption[] options)
-      : base() {
+  protected BaseWindowRenderer(UnityEngine.GUILayoutOption[] options) {
     UnityEngine.GUILayoutOption[] default_options = {GUILayoutMinWidth(20)};
     options_ = options.Length == 0 ? default_options : options;
-    lock_name_ = GetType().ToString() + ":lock:" + GetHashCode();
+    lock_name_ = GetType() + ":lock:" + GetHashCode();
```
Comment (eggrobin) [inline ksp_plugin_adapter/window_renderer.cs]: `$"{GetType()}:lock:{GetHashCode()}"`

File: ksp_plugin_adapter/window_renderer.cs [hunk 8]
```diff
@@ -142,7 +137,7 @@ protected BaseWindowRenderer(UnityEngine.GUILayoutOption[] options)
                 id         : GetHashCode(),
                 screenRect : new UnityEngine.Rect(
                     x      : (UnityEngine.Screen.width - rectangle_.width) / 2,
-                    y      : UnityEngine.Screen.height / 3,
+                    y      : (float)UnityEngine.Screen.height / 3,
```
Comment (eggrobin) [inline ksp_plugin_adapter/window_renderer.cs]: Why?

File: ksp_plugin_adapter/ksp_plugin_adapter.cs (hunk 4 above)
Comment (eggrobin) [reply ksp_plugin_adapter/ksp_plugin_adapter.cs]: Let us go with BactrianCamelCase but noun phrases.

File: ksp_plugin_adapter/window_renderer.cs (hunk 8 above)
Comment (eggrobin) [reply ksp_plugin_adapter/window_renderer.cs]: `/ 3f` (and comment than `y` is float so `3.0` would double-round.

File: ksp_plugin_adapter/ksp_plugin_adapter.cs (hunk 4 above)
Comment (pleroy) [reply ksp_plugin_adapter/ksp_plugin_adapter.cs]: Done, but I am unconvinced.

**Comment by pleroy** [comment]
retest this please

**Pull request merged**